### DISCUSSION TITLE
Support for non-Nvidia GPUs hwaccel codecs

### DISCUSSION THREAD

**Discussion by RChasedev** [rustdesk/rustdesk#9572] [General]
Thank you for this great project! It would be great to have support for all three major brands of GPU for hardware accelerated codecs (Nvidia, AMD, Intel). Suggest VAAPI libraries as a way to add both AMD and Intel features on Linux. Other future GPU targets might include MALI for phones and other embedded devices.

**Comment by Roquelobster** [comment]
I have a VAAPI GPU (Intel B580) in Linux that supports hardware encoding for AV1, h264 and h265 and only decoding for VP8 and VP9 (as per the table for the BMG family in https://github.com/intel/media-driver).

Looking at /rustdesk/blob/master/res/vcpkg/ffmpeg/portfile.cmake it would appear only linux-arm has VAAPI enabled and linux-x64 only has CUDA enabled.  Is it possible to add VAAPI hardware encoding/decoding to x64?

I see in my syslog Rustdesk entries complaints of libcuda not being able to be loaded and no mention of VAAPI.

Handbrake, VLC and mplayer all seemed to start using VAAPI once I had installed the libvdpau-va-gl1  library as they seemed to only use VDPAU.  The portfile.cmake has VDPAU disabled.
